app/github_repo_fetcher.rb

```diff
@@ -34,6 +34,30 @@ def docs(repo_name)
     end
   end
 
+  def nest_docs(flat_docs)
+    nested_docs = { docs: [] }
+    flat_docs.each do |doc|
+      subdirs = doc[:path].match(/apps\/(?:[^\/])+\/(.+)/)[1].split("/")
+      subdirs.pop # to remove the filename - we only care about directories
+
+      hash_to_inject_into = nested_docs
+      subdirs_processed = 0
+      while subdirs_processed < subdirs.count
+        title = "#{subdirs[subdirs_processed]}/"
+        exists = hash_to_inject_into[:docs].find { |elm| elm[:title] == title }
+        hash_to_inject_into[:docs] << { title: title, docs: [] } unless exists
+        hash_to_inject_into = hash_to_inject_into[:docs].find { |elm| elm[:title] == title }
+        subdirs_processed += 1
+      end
```

Comment:
I wonder if it's worth extracting this into a function such as `find_parent_hash`, that would mean we could use a `for` loop on `subdirs` with a `return`, which I think might be a bit nicer than the `while subdirs_processed < subdirs.count` approach?

Reply:
This was easier said than done! I think I've improved it now 👍